### ISSUE TITLE
Add a generated stringification method

### ISSUE THREAD

**Issue by staktrace** [eqrion/cbindgen#580]
I'd like to add the ability for cbindgen to generate a stringifier for enums and structs. It could be controlled by a `derive-ostream` option similar to `derive-eq` and friends. For enums it would generate something like:

```
std::ostream& operator<<(std::ostream& aStream, const MyEnum& aEnum) {
  switch (aEnum) {
    case One:
      aStream << "One";
      break;
    ...
  }
  return aStream;
}
```

and for structs it would do something like this:

```
struct MyStruct {
  int x;
  int y;
  ...
  friend std::ostream& operator<<(std::ostream& aStream, const MyStruct& aStruct) {
    return aStream << "{ x=" << aStruct.x << ", y=" << aStruct.y << " }";
  }
  ...
};
```

Any objections/suggestions? The goal here is to remove the hand-written [stuff](https://searchfox.org/mozilla-central/rev/f27594d62e7f1d57626889255ce6a3071d67209f/gfx/layers/wr/WebRenderLayersLogging.cpp#14-70) that we have in m-c as part of bug the cleanup I'm doing in bug 1666802 and friends.

**Comment by emilio** [comment]
Seems sensible to me. ADTs may be a bit funnier but nothing too hard I think :)

**Comment by staktrace** [comment]
This is fixed by #582 and followups.
